test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java

```diff
@@ -37,8 +48,8 @@ public class MessageProducerClient implements AutoCloseable {
     public void startProducer(final String topicName) {
 
         try {
-            final ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(QUEUE_URI);
-            connection = factory.createConnection();
+            activeMQConnectionFactory.setBrokerURL(QUEUE_URI);
```

Comment:
Setting QUEUE_URI through constructor is wrapping this statement in a try catch and throwing IllegalStateException (Inside ActiveMQConnectionFactory class), not entirely sure what exactly is happening in this setBrokerURL, may be worth replicating it here as well

Reply:
This is what is happening in setBrokerURL()

if (this.readOnly) {
            throw new javax.jms.IllegalStateException("You cannot use setBrokerURL after the connection factory has been used");
        }

Reply:
It's to stop people starting the connection then changing the url, which would be daft. However, we've not started the connection factory yet in out code, so it's academic